apps/web/src/pages/auth/components/HubspotSignupForm.tsx

```diff
@@ -0,0 +1,134 @@
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { useMutation } from '@tanstack/react-query';
+import decode from 'jwt-decode';
+
+import { JobTitleEnum } from '@novu/shared';
+import type { ProductUseCases, IResponseError, ICreateOrganizationDto, IJwtPayload } from '@novu/shared';
+
+import { api } from '../../../api/api.client';
+import { useAuthContext } from '../../../components/providers/AuthProvider';
+import { useVercelIntegration, useVercelParams } from '../../../hooks';
+import { ROUTES } from '../../../constants/routes.enum';
+import { HubspotForm } from '@novu/shared-web';
+import { HUBSPOT_FORM_IDS } from '../../../constants/hubspotForms';
+import SetupLoader from './SetupLoader';
+
+export function HubspotSignupForm() {
+  const [loading, setLoading] = useState<boolean>();
+  const navigate = useNavigate();
+  const { setToken, token, currentUser } = useAuthContext();
+  const { startVercelSetup } = useVercelIntegration();
+  const { isFromVercel } = useVercelParams();
+
+  const { mutateAsync: createOrganizationMutation } = useMutation<
+    { _id: string },
+    IResponseError,
+    ICreateOrganizationDto
+  >((data: ICreateOrganizationDto) => api.post(`/v1/organizations`, data));
+
+  useEffect(() => {
+    if (token) {
+      const userData = decode<IJwtPayload>(token);
+
+      if (userData.environmentId) {
+        if (isFromVercel) {
+          startVercelSetup();
+
+          return;
+        }
+
+        navigate(ROUTES.HOME);
+      }
+    }
+  }, [token, navigate, isFromVercel, startVercelSetup]);
+
+  async function createOrganization(data: IOrganizationCreateForm) {
+    const { organizationName, jobTitle, ...rest } = data;
+    const createDto: ICreateOrganizationDto = { ...rest, name: organizationName, jobTitle };
+    const organization = await createOrganizationMutation(createDto);
+    const organizationResponseToken = await api.post(`/v1/auth/organizations/${organization._id}/switch`, {});
+
+    setToken(organizationResponseToken);
+  }
+
+  function jwtHasKey(key: string) {
+    if (!token) return false;
+    const jwt = decode<IJwtPayload>(token);
+
+    return jwt && jwt[key];
+  }
+
+  const onCreateOrganization = async (data: IOrganizationCreateForm) => {
```

Comment:
Small naming suggestion to better communciate intent.

```suggestion
  const handleCreateOrganization = async (data: IOrganizationCreateForm) => {
```